test/vote/vote.model.spec.js

```diff
@@ -0,0 +1,94 @@
+// vote.model.spec.js
+// Unit tests on the vote model
+"use strict";
+
+var mongoose = require('mongoose');
+
+
+var Vote = require(process.cwd() + '/src/vote/vote.model');
+
+var helperFile = require('../helper');
+let cleanDatabase = helperFile.cleanDatabase;
+
+
+var assert = require('chai').assert;
+
+
+// Test the casting of votes
+describe('Casting a size', function () {
+
+  beforeEach(function (done) {
+    cleanDatabase(function () {
+      done();
+    });
+  });
+
+  // We cast a vote of size 2
+  // At the end we expect to be given a vote object with the same
+  // details as we said
+  it('castVote, size 2 selected - not on a real story', function (done) {
+    // This should fail - vote must be dependant on an actual story!
+    var storyId = "dummy story id - not a mongo ID!!!"
+
+    Vote.create({
+      story: storyId,
+      size: 2
+    }, function (err, theVote) {
+      assert.isNotNull(err, "We should not accept bogus story ids")
+      done()
+    });
+
+  });
+
+
+  it('castVote, size 1 selected - on a real story', function (done) {
+    // This should fail - vote must be dependant on an actual story!
+
+    let request = helperFile.request;
+
+    request()
+      .post('/stories')
```

Comment:
Since `Vote` depends on `Story` we need to feed the database with a story first in order to vote for it.

According to @Zsuark depending on a public API would be safer that using the private models. So that is because we populated the database with a `post` request here.

On the other hand. I think that we should not to depend on upper layers in order to test models. After all, we are testing a validation rule belonging to the persistence layer.

What do you think? Is this related to you question?